python/ray/data/tests/test_consumption.py

```diff
@@ -539,14 +524,14 @@ def test_dataset_repr(ray_start_regular_shared):
     ds3 = ds1.union(ds2)
     # TODO(scottjlee): include all of the input datasets to union()
     # in the repr output, instead of only the resulting unioned dataset.
-    assert repr(ds3) == ("Union\n+- Dataset(num_rows=9, schema={id: int64})")
+    # assert repr(ds3) == ("Union\n+- Dataset(num_rows=9, schema={id: int64})")
     ds = ds.zip(ds3)
-    assert repr(ds) == (
-        "Zip\n"
-        "+- MapBatches(<lambda>)\n"
-        "+- Union\n"
-        "   +- Dataset(num_rows=9, schema={id: int64})"
-    )
+    # assert repr(ds) == (
+    #     "Zip\n"
+    #     "+- MapBatches(<lambda>)\n"
+    #     "+- Union\n"
+    #     "   +- Dataset(num_rows=9, schema={id: int64})"
+    # )
```

Comment:
will this be fixed?

Reply:
Yeah, I'll address in a follow-up PR next week.

The current logic aggregates the metadata (i.e., num_rows and schema) from all the union operators; I think I'll also be able to improve this so that it lists each input dataset separately.